File: src/main/kotlin/com/itmo/microservices/demo/order/impl/entity/CatalogItem.java
package com.itmo.microservices.demo.order.impl.entity;

import lombok.*;
import org.hibernate.Hibernate;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import java.util.Objects;
import java.util.UUID;

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
public class CatalogItem {
    @Id
    @GeneratedValue
    private UUID uuid;

    private String title;
    private String description;
    private int price;
    private int amount;

    @ManyToOne
    private Order order;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        CatalogItem that = (CatalogItem) o;
        return uuid != null && Objects.equals(uuid, that.uuid);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: src/main/kotlin/com/itmo/microservices/demo/order/impl/entity/CatalogItem.kt
package com.itmo.microservices.demo.order.impl.entity

import lombok.Getter
import lombok.RequiredArgsConstructor
import lombok.Setter
import lombok.ToString
import org.hibernate.Hibernate
import java.util.*
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToOne

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
class CatalogItem {
    @Id
    @GeneratedValue
    private val uuid: UUID? = null
    private val title: String? = null
    private val description: String? = null
    private val price = 0
    private val amount = 0

    @ManyToOne
    private val order: Order? = null
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false
        val that = o as CatalogItem
        return uuid != null && uuid == that.uuid
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}